app/renderer/components/Inspector/HighlighterRect.js

```diff
@@ -1,38 +1,55 @@
 import React, { Component } from 'react';
 import InspectorCSS from './Inspector.css';
-import { parseCoordinates } from './shared';
 
 /**
  * Absolute positioned divs that overlay the app screenshot and highlight the bounding
  * boxes of the elements in the app
  */
+
 export default class HighlighterRect extends Component {
 
   render () {
-    const {selectedElement = {}, selectHoveredElement, unselectHoveredElement, hoveredElement = {}, selectElement, unselectElement, element,
-           zIndex, scaleRatio, xOffset, elLocation, elSize} = this.props;
+    const {selectedElement = {}, selectHoveredElement, unselectHoveredElement, hoveredElement = {}, selectElement, unselectElement, element, elementProperties,
+           scaleRatio, xOffset, elLocation, elSize, renderElementBoxes } = this.props;
     const {path: hoveredPath} = hoveredElement;
     const {path: selectedPath} = selectedElement;
 
-    let width, height, left, top, highlighterClasses, key;
+    let highlighterClasses, centroidClasses, key, divStyle;
+    let {left, top, width, height, centerX, centerY} = elementProperties.dimensions;
     highlighterClasses = [InspectorCSS['highlighter-box']];
+    centroidClasses = [InspectorCSS['centroid-box']];
 
     if (element) {
-      // Calculate left, top, width and height coordinates
-      const {x1, y1, x2, y2} = parseCoordinates(element);
-      left = x1 / scaleRatio + xOffset;
-      top = y1 / scaleRatio;
-      width = (x2 - x1) / scaleRatio;
-      height = (y2 - y1) / scaleRatio;
 
       // Add class + special classes to hovered and selected elements
       if (hoveredPath === element.path) {
-        highlighterClasses.push(InspectorCSS['hovered-element-box']);
+        renderElementBoxes ?
+          highlighterClasses.push(InspectorCSS['hovered-element-box'])
+          :
+          centroidClasses.push(InspectorCSS['hovered-element-box'])
+          &&
+          highlighterClasses.push(InspectorCSS['centroid-hovered-element-box']);
       }
       if (selectedPath === element.path) {
-        highlighterClasses.push(InspectorCSS['inspected-element-box']);
+        renderElementBoxes ?
+          highlighterClasses.push(InspectorCSS['inspected-element-box'])
+          :
+          centroidClasses.push(InspectorCSS['inspected-element-box']);
       }
-      key = element.path;
+
+      key = renderElementBoxes ? element.path : `C:${element.path}`;
+
+      divStyle = !renderElementBoxes ?
+        {
+          left: (centerX - (10 / scaleRatio + (xOffset || 0)) || 0),
+          top: (centerY - (10 / scaleRatio) || 0),
+          width: (20 / scaleRatio),
+          height: (20 / scaleRatio),
+          ...(elementProperties.container && { borderRadius: '50%' }),
```

Comment:
not sure if spread operator would work here if elementProperties.container is falsy